boot-microservices/src/main/java/org/ow2/proactive/boot/microservices/iam/util/IAMSessionUtil.java

```diff
@@ -0,0 +1,150 @@
+
+        tokenSignatureKey = config.getString(IAMConfiguration.IAM_TOKEN_SIGNATURE_KEY);
+
+        tokenEncryptionKey = config.getString(IAMConfiguration.IAM_TOKEN_ENCRYPTION_KEY);
+
+        isJWTSession = config.getBoolean(IAMConfiguration.PA_REST_SESSION_AS_JWT);
+
+        ticketMarker = config.getString(IAMConfiguration.SSO_TICKET_MARKER);
+    }
+
+    /**
+     * indicates whether the format of IAM sessions is JWT (Json Web Token) or not.
+     *
+     * @return boolean indicating whether IAM sessions are generated as JWT.
+     * @since version 8.3.0
+     */
+    public boolean isJWTSession() {
+        return isJWTSession;
+    }
+
+    /**
+     * deletes IAM sessions by destroying SSO tickets registered in IAM microservice.
+     *
+     * @param ssoTicket SSO ticket to be destroyed
+     * @return boolean indicating whether the SSO ticket is destroyed
+     * @since version 8.3.0
+     */
+    public boolean deleteSession(String ssoTicket) {
+        return new IAMRestClient().deleteSSOTicket(iamURL + IAMConfiguration.IAM_TICKET_REQUEST + "/" + ssoTicket);
+    }
+
+    /**
+     * indicates whether the IAM session is still valid.
+     *
+     * @param ssoTicket SSO ticket of IAMSession to be validated
+     * @return boolean indicating whether IAM session is valid.
+     * @since version 8.3.0
+     */
+    public boolean sessionIsValid(String ssoTicket) {
+        return new IAMRestClient().isSSOTicketValid(iamURL + IAMConfiguration.IAM_TICKET_REQUEST + "/" + ssoTicket);
+    }
+
+    /**
+     * creates a new IAM session given user credentials.
+     *
+     * @param username user login
+     * @param password user password
+     * @return a key-value pair, the key represents the session ID, and the value are JWT claims (when the session token is generated as a JWT).
+     * @since version 8.3.0
+     */
+    public AbstractMap.SimpleEntry<String, JwtClaims> createNewSessionToken(String username, char[] password) {
+
+        String sessionToken = new IAMRestClient().getSSOTicket(iamURL + IAMConfiguration.IAM_TICKET_REQUEST,
+                                                               username,
+                                                               password,
+                                                               ticketMarker,
+                                                               isJWTSession);
+
+        JwtClaims claims = isJWTSession ? JWTUtils.parseJWT(sessionToken,
+                                                            tokenSignatureEnabled,
+                                                            tokenSignatureKey,
+                                                            tokenEncryptionEnabled,
+                                                            tokenEncryptionKey,
+                                                            iamURL,
+                                                            iamURL)
+                                        : null;
```

Comment:
Is there any other value that we could store than null? Because then, it will be the value of a map record, and when searching by key afterwards we won't know whether there is a key and its value is null or if there is not such key in the map.

Reply:
When the value of the map is null, no search can be performed on it, and a NullPointerException will be thrown. I did it in purpose as a 'fail-fast' mode to avoid using this map (JwtClaims) when the session id is not a JWT (json web token).